File: frontend/types/transaction.ts
export type TransactionKind =
  | "funding"
  | "transfer"
  | "airtime"
  | "bill_payment";

export type TransactionDirection = "credit" | "debit";
export type TransactionStatus = "successful" | "failed" | "pending";

export type Transaction = {
  id: string;
  reference: string;
  sender_id?: string | null;
  receiver_id?: string | null;
  amount: number;
  transaction_type: TransactionKind;
  direction: TransactionDirection;
  status: TransactionStatus;
  description?: string | null;
  created_at: string;
};

export type Beneficiary = {
  id: string;
  user_id: string;
  beneficiary_name: string;
  account_number: string;
  phone_number?: string | null;
  bank_name?: string | null;
  created_at: string;
};

File: backend/app/schemas/transaction.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import TransactionDirection, TransactionStatus, TransactionType


class TransactionResponse(BaseModel):
    id: str
    reference: str
    sender_id: str | None
    receiver_id: str | None
    amount: Decimal
    transaction_type: TransactionType
    direction: TransactionDirection
    status: TransactionStatus
    description: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
